src/ucs/vfs/base/vfs_obj.c

```diff
@@ -259,13 +259,21 @@ static ucs_status_t ucs_vfs_node_add(void *parent_obj, ucs_vfs_node_type_t type,
                                      va_list ap, ucs_vfs_node_t **new_node)
 {
     ucs_vfs_node_t *current_node;
-    char rel_path_buf[PATH_MAX];
-    char abs_path_buf[PATH_MAX];
+    char *rel_path_buf = ucs_malloc(PATH_MAX, "rel_path_buf");
+    char *abs_path_buf = ucs_malloc(PATH_MAX, "abs_path_buf");
     char *token, *next_token;
+    ucs_status_t status = UCS_OK;
+
+    if ((rel_path_buf == NULL) || (abs_path_buf == NULL)) {
+        ucs_error("Failed to allocate memory for path buffers");
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
```

Comment:
remote `out` label and use `return UCS_ERR_NO_MEMORY` instead, rename `out_free_buffers` to `out`

Reply:
return here.

Reply:
there could be a case where one of the buffer was allocated successfully and the other one failed, so I think we still need to free, so I changed it to "goto out", let me know what do you think

Reply:
Outdated, done in other PRs